### PR TITLE
SQL input mode classification fix

### LINKED ISSUE #532: latest release broke my query
**Describe the bug**
release 0.48.11 broke my query which has always worked before.

**To Reproduce**

```
$ sq sql --src=@log 'select*from log'
0 rows affected
```

**Expected behavior**

sql to output results as in version 0.48.10 and earlier

**`sq` version**

Paste the output of `sq version --yaml` into the code block below:

```yaml
version: v0.48.11
commit: ecd7ad0
timestamp: "2026-01-18T20:46:08Z"
latest_version: v0.48.11
host:
  platform: linux
  arch: amd64
  kernel: Linux
  kernel_version: 6.18.2-x64
  variant: Arch Linux
```

**Additional context**

now I have to use `sq sql --src=@log 'select *from log'` to get results.

### PR DESCRIPTION
**Pull request by neilotoole**
## Summary

- Replaces the custom `isQueryString` implementation in `DefaultExecModeFor` with 
  calls to the usql library's `FindPrefix` and `QueryExecType` functions
- The usql library provides battle-tested SQL classification used across 40+ 
  database drivers

## Changes

- Add `github.com/xo/usql` dependency (only `stmt` and `drivers` packages used)
- Rewrite `DefaultExecModeFor` to use usql's `FindPrefix` and `QueryExecType`
- Remove the custom `isQueryString` function
- Update tests with comprehensive coverage for queries, DDL/DML statements, 
  and edge cases

## Benefits

- Support for 200+ SQL statement keywords (vs ~10 in previous implementation)
- Handles special cases like `SELECT INTO`, `PRAGMA`, and CREATE/DROP modifiers
- Proper detection of compact SQL patterns (`SELECT*`, `SELECT(1)`, `SELECT-1`, etc.)
- Maintained upstream library inherits future improvements

Fixes #532

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -71,18 +71,23 @@ require (
 	golang.org/x/text v0.33.0
 )
 
+require github.com/xo/usql v0.20.8
```
[the remark this answers is not in the document]
Comment (neilotoole) [reply go.mod]: Fixed

**Review by drluckyspin — DISMISSED** [review]
This is great! Major upgrade.
